pkg/syncer/index.go

```diff
@@ -62,15 +63,20 @@ func (i ChartIndex) Get(id string) *Chart {
 }
 
 // loadCharts loads the charts map into the index from the source repo
-func (s *Syncer) loadCharts(charts ...string) error {
+func (s *Syncer) loadCharts(charts ...string) (bool, error) {
 	if len(charts) == 0 {
 		if !s.autoDiscovery {
-			return errors.Errorf("unable to discover charts to sync")
+			return true, errors.Errorf("unable to discover charts to sync")
 		}
-
 		srcCharts, err := s.cli.src.List()
+		// For OCI source we need either access to a charts index file in the repo or a list of charts provided via
+		// config file
+		if len(srcCharts) == 0 && s.source.GetRepo().GetKind() == api.Kind_OCI {
```

Comment:
I don't understand what has changed that makes it different than before? 

Is it so we can detect if the index was empty?

Reply:
Two things:
- the oci.List() method can return an empty list of charts as per jdrios comment
- we can now detect which errors coming from loadCharts function should abort the execution and which don't. Previously all errors were reported at the end but charts-syncer ended with exit code 0.